src/Npgsql/TypeMapping/ConnectorTypeMapper.cs

```diff
@@ -413,19 +413,26 @@ void BindRangeType(NpgsqlTypeHandler elementHandler, PostgresRangeType pgRangeTy
             if (TryGetMapping(postgresType, out var npgsqlTypeMapping))
                 return (npgsqlTypeMapping.NpgsqlDbType, postgresType);
 
-            // Try to find the Elements' postgresType in the mappings
+            // Try to find the elements' postgresType in the mappings
             if (postgresType is PostgresArrayType arrayType &&
                 TryGetMapping(arrayType.Element, out var elementNpgsqlTypeMapping))
-            {
                 return (elementNpgsqlTypeMapping.NpgsqlDbType | NpgsqlDbType.Array, postgresType);
-            }
+
+            // Try to find the elements' postgresType of the base type in the mappings
+            // this happens with domains over arrays
+            if (postgresType is PostgresDomainType domainType && domainType.BaseType is PostgresArrayType baseType &&
+                TryGetMapping(baseType.Element, out var baseTypeElementNpgsqlTypeMapping))
+                return (baseTypeElementNpgsqlTypeMapping.NpgsqlDbType | NpgsqlDbType.Array, postgresType);
 
             // It might be an unmapped enum/composite type, or some other unmapped type
             return (null, postgresType);
 
             bool TryGetMapping(PostgresType pgType, out NpgsqlTypeMapping mapping)
-                => (Mappings.TryGetValue(pgType.Name, out mapping) ||
-                    Mappings.TryGetValue(pgType.FullName, out mapping));
+                => Mappings.TryGetValue(pgType.Name, out mapping) ||
```

Comment:
The opening bracket padded the expression to align it with the next one. I think we should keep it.

Reply:
Hmmm...
I would't keep it just for this.
How about manually padding the following rows?
At least on my system (VS 2017 + Resharper) the manual padding is kept intact.

Reply:
It's okay too for me. Thanks!